src/output-tx.c

```diff
@@ -138,35 +148,87 @@ static TmEcode OutputTxLog(ThreadVars *tv, Packet *p, void *thread_data, PacketQ
         if (!(AppLayerParserStateIssetFlag(f->alparser, APP_LAYER_PARSER_EOF)))
         {
             int tx_progress = AppLayerParserGetStateProgress(p->proto, alproto,
-                                                             tx, FlowGetDisruptionFlags(f, STREAM_TOSERVER));
-            if (tx_progress < tx_progress_done_value_ts) {
-                SCLogDebug("progress not far enough, not logging");
-                break;
+                tx, FlowGetDisruptionFlags(f, STREAM_TOSERVER));
+            if (tx_progress == tx_progress_done_value_ts) {
+                ts_ready = 1;
             }
 
             tx_progress = AppLayerParserGetStateProgress(p->proto, alproto,
-                                                         tx, FlowGetDisruptionFlags(f, STREAM_TOCLIENT));
-            if (tx_progress < tx_progress_done_value_tc) {
-                SCLogDebug("progress not far enough, not logging");
-                break;
+                tx, FlowGetDisruptionFlags(f, STREAM_TOCLIENT));
+            if (tx_progress == tx_progress_done_value_tc) {
+                tc_ready = 1;
             }
         }
 
+        if (!(ts_ready || tc_ready)) {
+            SCLogNotice("progress not for enough, not logging");
+            break;
+        }
+
+        /* Prevent the processing of one direction getting farther
```

Comment:
On 17-12-15 16:53, Jason Ish wrote:
> In src/output-tx.c
> <https://github.com/inliniac/suricata/pull/1794#discussion_r47921644>:
> 
>>              }
>>          }
>>  
>> +        if (!(ts_ready || tc_ready)) {
>> +            SCLogNotice("progress not for enough, not logging");
>> +            break;
>> +        }
>> +
>> +        /* Prevent the processing of one direction getting farther
> 
> I guess pipelining is different than the case I present, as everything
> should be order, so it could be handled by the current tracking. I think
> this would be pretty easy to extend to that case when we have a logger
> that would make use of it - like logging HTTP requests prior to the reply.

If we have:

REQ 1
REQ 2
REQ 3
RESP 1
RESP 2
RESP 3

Could we in the current patch log all the requests before the responses
are received?